src/Simulation/CsharpGeneration/AssemblyInfo.fs

```diff
@@ -0,0 +1,11 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace QsCompiler.AssemblyInfo
+
+open System.Runtime.CompilerServices
+
+[<assembly: InternalsVisibleTo("Microsoft.Quantum.CsharpGeneration.App" + SigningConstants.PUBLIC_KEY)>]
```

Comment:
why are these necessary? In particular, I would make sure that anything that `CsharpGeneration.App` needs is public as it will probably be needed by iq#.